.github/workflows/build.yml

```diff
@@ -55,18 +49,103 @@ jobs:
+            cd bloomberg-comdb2/build &&
             cmake .. &&
-            make &&
-            sudo make install
+            make
           )
         '
+      - name: Archive bloomberg-comdb2 repo with build artifacts
+        run: 'tar czvf bloomberg-comdb2.tar.gz bloomberg-comdb2/'
+      - name: Upload bloomberg-comdb2 repo with build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: bloomberg-comdb2
+          path: ./bloomberg-comdb2.tar.gz
+
+  build_sdist:
+    name: Build python-comdb2 source distribution
+    runs-on: ubuntu-latest
+    needs: [build_bloomberg_comdb2]
+    steps:
+      - name: Download bloomberg-comdb2 with build artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: bloomberg-comdb2
+          path: .
+      - name: Install bloomberg-comdb2
+        run: '
+          sudo apt-get install -qy
+            libevent-dev
+            liblz4-dev
+            libprotobuf-c-dev
+            libsqlite3-dev
+            libssl-dev
+            libunwind-dev
+            zlib1g-dev &&
+          tar xvf bloomberg-comdb2.tar.gz &&
+          (cd bloomberg-comdb2/build && sudo make install) &&
+          rm -rf bloomberg-comdb2.tar.gz bloomberg-comdb2/
+        '
+      - uses: actions/checkout@v4
+      - name: 'Set up Python 3.7'
+        uses: actions/setup-python@v4  # note that this step overwrites the PKG_CONFIG_PATH variable
+        with:
+          python-version: '3.7'  # the lowest version that we support in CI
+      - name: Build sdist
+        run: '
+          sudo apt-get install -qy pkg-config &&
+          PKG_CONFIG_PATH=/opt/bb/lib/pkgconfig:/usr/lib/x86_64-linux-gnu/pkgconfig
+          pipx run build --sdist
+        '
+      - name: Upload sdist
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-comdb2-sdist
+          path: dist/*.tar.gz
+
+  test:
+    name: Test suite
+    runs-on: ubuntu-latest
+    needs: [build_sdist]
+    strategy:
+      matrix:
+        python-version:
+          - '3.12'
+          - '3.11'
+          - '3.10'
+          - '3.9'
+          - '3.8'
+          - '3.7'
+    steps:
+      - name: Download comdb2 repo with build artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: bloomberg-comdb2
+          path: .
+      - name: Install bloomberg-comdb2
+        run: '
+          sudo apt-get install -qy
+            libevent-dev
+            liblz4-dev
+            libprotobuf-c-dev
+            libsqlite3-dev
+            libssl-dev
+            libunwind-dev
+            zlib1g-dev &&
+          tar xvf bloomberg-comdb2.tar.gz &&
+          (cd bloomberg-comdb2/build && sudo make install) &&
+          rm -rf bloomberg-comdb2.tar.gz bloomberg-comdb2/
+        '
       - name: Start local comdb2 instance
         run: '
           sudo mkdir -p /opt/bb/share/schemas/$COMDB2_DBNAME &&
+          sudo mkdir -p /opt/bb/etc/cdb2/config &&
```

Comment:
Huh, it's odd that this wasn't required before and now is

Reply:
Hmm… I'm not sure why this was anymore, probably because it is added by `sudo make install`, but I tried to make this PR work with a .tar.gz archive of `/opt/bb`… This may not be needed, I'll try to remove it.

Reply:
Having it is fine if we need it, but I can't immediately spot a reason why it ought to be required.